[RFC] vim-patch:7.4.771

jbradaric opened the pull request:
```
Problem:    Search does not handle multi-byte character at the start position
            correctly.
Solution:   Take byte size of character into account. (Yukihiro Nakadaira)
```

https://github.com/vim/vim/commit/5f1e68b7bc241118e5dd8fc781147fdda881ada8

jbradaric committed:
vim-patch:7.4.771

Problem:    Search does not handle multi-byte character at the start position
            correctly.
Solution:   Take byte size of character into account. (Yukihiro Nakadaira)

https://github.com/vim/vim/commit/5f1e68b7bc241118e5dd8fc781147fdda881ada8

jbradaric committed:
Add test files for patch 7.4.771.

vim-patch:efcabd6892ad89a4585fb77aa94c3b1802b784ab

jbradaric committed:
search: Fix linter errors.

justinmk commented:
For the "Add tests files" commit, please put `vim-patch:efcabd6892ad89a4585fb77aa94c3b1802b784ab` in the first line. This allows `scripts/vim-patch.sh -l` to detect it.

justinmk commented on test/functional/legacy/search_mbyte_spec.lua:
```diff
@@ -0,0 +1,31 @@
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('search_mbyte', function()
+  setup(clear)
```
Usually I recommend `before_each(clear)` so that if new `it()` blocks are added in the future, it won't confuse people. (`setup()` runs only once per `describe()` block, whereas `before_each()` runs before each `it()`).

jbradaric replied:
@justinmk Thanks, I've replaced `setup` with `before_each`.

Btw, the `legacy2luatest.pl` script generates tests with the `setup(clear)` line. Should this also be changed or noted somewhere?

justinmk replied:
Yes, could you make that change here?

jbradaric replied:
@justinmk I've updated `legacy2luatest.pl` to use `before_each` instead of `setup`.

jbradaric committed:
Use before_each instead of setup in the test.

jbradaric committed:
Remove unused variables from the test.

jbradaric committed:
legacy2luatest: Use before_each instead of setup.

This will avoid confusion if additional `it()` blocks are added.
(`setup()` only runs once per `describe()` block, whereas `before_each()`
runs before each `it()`).

justinmk commented on test/functional/legacy/search_mbyte_spec.lua:
```diff
@@ -0,0 +1,31 @@
+-- Test for search('multi-byte char', 'bce')
```
these comments can go in the `describe()` or `it()` descriptions instead. I'll fix this when merging so we don't hold up this PR any longer.

justinmk commented on test/functional/legacy/search_mbyte_spec.lua:
```diff
@@ -0,0 +1,31 @@
+      Ａ]=])
+
+    execute('source small.vim')
+    execute('source mbyte.vim')
```
these files don't exist, so I removed these lines.
